be/src/vec/sink/vdata_stream_sender.cpp

```diff
@@ -266,7 +266,13 @@ Status Channel::close_internal() {
         status = send_current_block(true);
     } else {
         SCOPED_CONSUME_MEM_TRACKER(_parent->_mem_tracker.get());
-        status = send_block((PBlock*)nullptr, true);
+        if (is_local()) {
+            if (_recvr_is_valid()) {
```

Comment:
if just remove the sender not send a null block to receiver. Then the receiver may not get eof signal?

Reply:
The receiver just judge the eof by `sender_number == 0`;